pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -194,13 +195,19 @@ func BuildListenerTLSContext(serverTLSSettings *networking.ServerTLSSettings,
 	}
 
 	// Set TLS parameters if they are non-default
-	if len(serverTLSSettings.CipherSuites) > 0 ||
+	var ecdhCurves []string
+	if mesh != nil && mesh.MeshExternal_TLS != nil &&
+		(serverTLSSettings.Mode == networking.ServerTLSSettings_MUTUAL || serverTLSSettings.Mode == networking.ServerTLSSettings_SIMPLE) {
+		ecdhCurves = mesh.MeshExternal_TLS.EcdhCurves
+	}
+	if len(serverTLSSettings.CipherSuites) > 0 || len(ecdhCurves) > 0 ||
 		serverTLSSettings.MinProtocolVersion != networking.ServerTLSSettings_TLS_AUTO ||
 		serverTLSSettings.MaxProtocolVersion != networking.ServerTLSSettings_TLS_AUTO {
 		ctx.CommonTlsContext.TlsParams = &auth.TlsParameters{
 			TlsMinimumProtocolVersion: convertTLSProtocol(serverTLSSettings.MinProtocolVersion),
 			TlsMaximumProtocolVersion: convertTLSProtocol(serverTLSSettings.MaxProtocolVersion),
 			CipherSuites:              serverTLSSettings.CipherSuites,
+			EcdhCurves:                ecdhCurves,
```

Comment:
Doubt why MeshExternal_TLS should be applied to listener

Reply:
I think it is needed for gateways receiving external traffic. Are you thinking we should only this for gateways and not for sidecar inbound?

Reply:
I think his point still valid - Should we only set when we are building listener for gateway and not sidecar?

Reply:
@ramaraochavali i think it is valid for external traffic still. We do have usecases where we donot use gateway, and uses sidercar TLS termination for external traffic using sidecar API

Reply:
But in this case it will also take effect on sidecar->sidecar communication

Reply:
yes. That is the side affect of this. We apply the same settings for cluster outbound. So while the setting says MeshExternal, in a way we are applying it for mesh calls also. That is the confusion I was talking about [here](https://github.com/istio/istio/pull/43207#discussion_r1112156882).

I agree though that it is very confusing implementation.

Reply:
I just added DNM label so that we can sort this out.

Reply:
Added this https://github.com/istio/api/pull/2543/files#r1112668620 in API.

Reply:
As per the discussions, API has been renamed to "tlsDefaults", instead of "external_tls". The implementation PR has been updated to use the new APIs